src/v/cluster/archival_metadata_stm.cc

```diff
@@ -83,36 +85,52 @@ struct archival_metadata_stm::snapshot
       envelope<snapshot, serde::version<1>, serde::compat_version<0>> {
     /// List of segments
     std::vector<segment> segments;
+    /// List of replaced segments
+    std::vector<segment> replaced;
     /// Start offset (might be different from the base offset of the first
-    /// segment)
+    /// segment). Default value means that the snapshot was old and didn't
+    /// have start_offset. In this case we need to set it to compute it from
+    /// segments.
     model::offset start_offset;
+    /// Last uploaded offset (default value means that old version was used
```

Comment:
Would be helpful to say which version we mean by "old version" (redpanda <= 22.2), so that future readers can understand.